service/storage_service.cc

```diff
@@ -1040,13 +1040,17 @@ storage_service::is_local_dc(const inet_address& targetHost) const {
 std::unordered_map<dht::token_range, std::vector<inet_address>>
 storage_service::get_range_to_address_map(const sstring& keyspace,
         const std::vector<token>& sorted_tokens) const {
+    if (keyspace != "") {
+        return construct_range_to_endpoint_map(keyspace, get_all_ranges(sorted_tokens));
+    }
     // some people just want to get a visual representation of things. Allow null and set it to the first
     // non-system keyspace.
-    if (keyspace == "" && _db.local().get_non_system_keyspaces().empty()) {
+    auto keyspaces = _db.local().get_non_system_keyspaces();
+
+    if (keyspaces.empty()) {
         throw std::runtime_error("No keyspace provided and no non system kespace exist");
     }
-    const sstring& ks = (keyspace == "") ? _db.local().get_non_system_keyspaces()[0] : keyspace;
-    return construct_range_to_endpoint_map(ks, get_all_ranges(sorted_tokens));
+    return construct_range_to_endpoint_map(keyspaces[0], get_all_ranges(sorted_tokens));
```

Comment:
It's nicer to have just one call, it allows the reader to verify that the only difference is how we get keyspace.

```c++
   if (keyspace.empty()) }
           // blah blah, check whatever
           keyspace = ...;
   }
   return construct_range_to_endpoint_map(keyspace, ...)
```

Reply:
I guess that was the original thought behind the original implementation.
`keyspace` is a const reference. it's not a critical path and it doesn't happen too often, so  I can use a string by value.
If you are ok with it, I'll change the implementation accordingly.

Reply:
String by value is fine here.